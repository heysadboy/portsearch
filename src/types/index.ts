import combineReducers from '../redux/reducers';

export interface IPort {
    code: string,
    name: string
}

export interface IMarketRate {
    day: string,
    mean: number | null,
    low: number | null,
    high: number | null
}

export interface IPointData {
    day: string,
    value: number | null
}

export interface ILineData {
    name: string,
    color: string,
    values: IPointData[]
}

export interface IGraphData {
    [key: string]: ILineData
}

export interface IPortAction {
    type: EPortActionType
    payload: IPort[]
}

export interface IMarketRateAction {
    type: EMarketRateActionType
    payload: IMarketRate[]
}

export enum EPortActionType {
    get_ports = "GET_PORTS"
}

export enum EMarketRateActionType {
    get_ports = "GET_MARKET_RATES"
}

export type AppState = ReturnType<typeof combineReducers>;
