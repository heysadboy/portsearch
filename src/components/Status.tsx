import { FC } from "react";
import { EStatusType } from "../types";

interface IStatusProps {
    status: EStatusType | string
}

const Status: FC<IStatusProps> = ({ status }) => {
    //Switch case to return the status based on the type
    //By passing type we can display different kind of statuses
    //If it is not a valid EStatusType then just string is displayed
    switch (status) {
        case EStatusType.error:
            return (
                <div className="ui error message">
                    <div className="header">Sorry, please try another route</div>
                    <p>Please select another route. We are unable to get market rates for this route.</p>
                </div>
            );
        case EStatusType.initial:
            return (
                <div className="ui info message">
                    <div className="header">Welcome to market rates by Xeneta</div>
                    <p>Please select the routes to see the information.</p>
                </div>
            );
        case EStatusType.loading:
            return (
                <div className="ui icon message">
                    <i className="notched circle loading icon"></i>
                    <div className="content">
                        <div className="header">Please wait for a moment</div>
                        <p>We're generating the graph for you.</p>
                    </div>
                </div>
            );
        case EStatusType.no_position_selected:
            return (
                <div className="ui warning message">
                    <div className="header">Can't generate graph without market position</div>
                    <p>Please select market position to see the graph.</p>
                </div>
            );
        case EStatusType.no_data:
            return (
                <div className="ui warning message">
                    <div className="header">Unable to process the data</div>
                    <p>No valid data exists between these route.</p>
                </div>
            );
        default:
            return (
                <div className="ui warning message">
                    <div className="header">{status}</div>
                </div>
            );;
    }
};

export default Status;