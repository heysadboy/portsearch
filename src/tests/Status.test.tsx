import TestRenderer from "react-test-renderer";
import Status from "../components/Status";
import { EStatusType } from "../types";

test("DataRange Component Testing.", () => {
    const component = TestRenderer.create(
        <Status status={EStatusType.error} />
    );

    let tree: any = component.toJSON();
    expect(tree).toMatchSnapshot();

    let errorDiv = component.root.findAllByType("div")[1];
    expect(errorDiv.children).toEqual(['Sorry, please try another route']);
});