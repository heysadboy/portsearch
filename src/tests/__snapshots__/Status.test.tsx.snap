// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataRange Component Testing. 1`] = `
<div
  className="ui error message"
>
  <div
    className="header"
  >
    Sorry, please try another route
  </div>
  <p>
    Please select another route. We are unable to get market rates for this route.
  </p>
</div>
`;
